WeakAurasTemplates/TriggerTemplatesData.lua

```diff
@@ -5051,32 +5586,40 @@ for regionType, regionData in pairs(WeakAuras.regionOptions) do
   end
 end
 
-if WeakAuras.IsClassic() then
-  -- consolidate talents from all specs in a new dummy "classic" spec, indexed by spell or title for no duplicate
-  for _, class in pairs(templates.class) do
-    class["classic"] = class["classic"] or {}
-    for specIndex, spec in pairs(class) do
-      if specIndex ~= "classic" then
-        for sectionIndex, section in pairs(spec) do
-          if not class["classic"][sectionIndex] then
-            class["classic"][sectionIndex] = {
-              icon = section.icon,
-              title = section.title,
-              args = {}
-            }
-          end
-          local args = class["classic"][sectionIndex].args
-          for itemIndex, item in pairs(section.args or {}) do
-            if item.spell then
-              args[item.spell] = item
-            else
-              args[itemIndex] = item
-            end
-          end
-        end
-      end
+-- remove from tables data for wrong version of the game
+for classId, classData in pairs(templates.class) do
+  for id, data in pairs(classData) do
+    if (tonumber(id) and WeakAuras.IsClassic())
+    or (id == "classic" and not WeakAuras.IsClassic())
```

Comment:
The structure of the templates looks hacky. As there's a classic table, but not a retail table. And those sections only exist in the class tables

Can you explain why you did it that way?

Reply:
I wanted to make it easier to re-use existing functions, but i'm re-organizing it to make it less hacky